immer/memory_policy.hpp

```diff
@@ -90,13 +91,15 @@ template <typename HeapPolicy,
           bool PreferFewerBiggerObjects =
               get_prefer_fewer_bigger_objects_v<HeapPolicy>,
           bool UseTransientRValues =
-              get_use_transient_rvalues_v<RefcountPolicy>>
+              get_use_transient_rvalues_v<RefcountPolicy>,
+          typename RelocationPolicy = no_relocation_policy>
 struct memory_policy
 {
     using heap       = HeapPolicy;
     using refcount   = RefcountPolicy;
     using transience = TransiencePolicy;
     using lock       = LockPolicy;
+    using relocation = RelocationPolicy;
```

Comment:
Since you are using this only as a on/off switch that then the other parts of the code use to actually implement the behaviour, maybe I'd prefer this to be a simple booal `bool enable_tracking_for_relocating_gc` or something alike with a comment maybe giving a couple more details on what this is trying to achieve.